src/main/java/task02/wrapper/PairListWrapper.java

```diff
@@ -0,0 +1,279 @@
+package task02.wrapper;
+
+import java.util.*;
+import java.util.stream.IntStream;
+
+public class PairListWrapper extends AbstractList<SimpleModel> {
+    private int defaultCapacity = 4;
+
+    private SimpleModel[] head = new SimpleModel[0];
+    private SimpleModel[] tail = new SimpleModel[0];
+
+    public PairListWrapper() {
+    }
+
+    public PairListWrapper(int defaultCapacity) {
+        this.defaultCapacity = defaultCapacity;
+    }
+
+    public int getDefaultCapacity() {
+        return defaultCapacity;
+    }
+
+    @Override
+    public SimpleModel get(int index) {
+        checkIndexBounds(index);
+        return getElementByIndex(index);
+    }
+
+    private SimpleModel[] getContainerByIndex(int index) {
+        return index < defaultCapacity ? head : tail;
+    }
+
+    @Override
+    public int size() {
+        return head.length + tail.length;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size() == 0;
+    }
+
+    @Override
+    public Iterator<SimpleModel> iterator() {
+        return new Iterator<>() {
+            private int position = 0;
+            private int prev = -1;
+
+            @Override
+            public boolean hasNext() {
+                return position < size();
+            }
+
+            @Override
+            public SimpleModel next() {
+                if (!hasNext()) {
+                    throw new NoSuchElementException();
+                }
+                SimpleModel result = getElementByIndex(position);
+                prev = position;
+                position++;
+                return result;
+            }
+
+            @Override
+            public void remove() {
+                if (prev < 0) {
+                    throw new IllegalStateException();
+                }
+                if (prev < defaultCapacity) {
+                    throw new NoChangeUnmodifiableException();
+                }
+                PairListWrapper.this.remove(prev);
+                if (prev < position) {
+                    position--;
+                    prev = -1;
+                }
+            }
+        };
+    }
+
+    @Override
+    public int indexOf(Object o) {
+        for (int i = 0; i < size(); i++) {
+            if (isSame(o, getElementByIndex(i))) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    @Override
+    public int lastIndexOf(Object o) {
+        for (int i = size() - 1; i >= 0; i--) {
+            if (isSame(o, getElementByIndex(i))) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    private SimpleModel getElementByIndex(int index) {
+        return index < defaultCapacity ? head[index] : tail[index - defaultCapacity];
+    }
+
+    private boolean isSame(Object o, SimpleModel obj) {
+        return Objects.isNull(obj) && Objects.isNull(o) || obj.equals(o);
```

Comment:
suggestion: you can use 
```suggestion
        Objects.equals(o, obj);
```